x-pack/plugins/maps/public/layers/vector_layer.d.ts

```diff
@@ -20,34 +21,51 @@ import { SyncContext } from '../actions/map_actions';
 
 type VectorLayerArguments = {
   source: IVectorSource;
-  joins: IJoin[];
+  joins?: IJoin[];
   layerDescriptor: VectorLayerDescriptor;
 };
 
 export interface IVectorLayer extends ILayer {
   getFields(): Promise<IField[]>;
   getStyleEditorFields(): Promise<IField[]>;
   getValidJoins(): IJoin[];
+  getSource(): IVectorSource;
 }
 
 export class VectorLayer extends AbstractLayer implements IVectorLayer {
   static createDescriptor(
-    options: Partial<VectorLayerDescriptor>,
+    options: Partial<LayerDescriptor>,
     mapColors?: string[]
   ): VectorLayerDescriptor;
 
   protected readonly _source: IVectorSource;
   protected readonly _style: IVectorStyle;
 
   constructor(options: VectorLayerArguments);
-
+  getLayerTypeIconName(): string;
   getFields(): Promise<IField[]>;
   getStyleEditorFields(): Promise<IField[]>;
   getValidJoins(): IJoin[];
+  _syncSourceStyleMeta(
+    syncContext: SyncContext,
+    source: IVectorSource,
+    style: IVectorStyle
+  ): Promise<void>;
+  _syncSourceFormatters(
+    syncContext: SyncContext,
+    source: IVectorSource,
+    style: IVectorStyle
+  ): Promise<void>;
+  syncLayerWithMB(mbMap: unknown): void;
   _getSearchFilters(
     dataFilters: MapFilters,
     source: IVectorSource,
     style: IVectorStyle
   ): VectorSourceRequestMeta;
   _syncData(syncContext: SyncContext, source: IVectorSource, style: IVectorStyle): Promise<void>;
+  ownsMbSourceId(sourceId: string): boolean;
```

Comment:
Should ownsMbSourceId and ownsMbLayerId be part of ILayer interface?

Reply:
yes, but I addressed this in https://github.com/elastic/kibana/pull/63533 . Would prefer to merge it as part of that PR, so this one can be without all the TS duct-tape.

Reply:
Yes please move to https://github.com/elastic/kibana/pull/63533